argo-cd-apps/base/vault.yaml

```diff
@@ -0,0 +1,33 @@
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: vault
+
+spec:
+  project: default
+
+  source:
+    path: components/vault
+    repoURL: https://github.com/redhat-appstudio/infra-deployments.git
+    targetRevision: main
+
+  destination:
+    namespace: spi-system
```

Comment:
spi-vault - then you do not need to set `namespace: spi-vault` in `components/vault/kustomization.yaml` (which I would rather have `components/spi-vault`)